lib/active_model_serializers/adapter/json_api/relationship.rb

```diff
@@ -34,6 +34,11 @@ def as_json
 
         private
 
+        def ensure_valid!(json)
```

Comment:
@remear yep - I am not saying the schema requires this stuff. I'm saying the docs require this stuff, and it looks like the docs and the schema are in conflict. So I opened up the issue on jsonapi to figure out which side has the bug.

> If you want to let consumers know that a relationship does exist but isn't available from the current available data, you'd probably provide a link for it, right?

This is the use case I would like to support. I would like to let consumers know the relationship does exist but isn't available from the current data. And I may not have a separate endpoint that I can link to - there's no reason for me to write a separate endpoint if my only use case for fetching this data is when I sideload.

> What if the relationship actually had meta if you were to fetch it?

Then we would render that `meta` as normal. We only render empty meta if no keys - including `meta` itself - are present.